dll/win32/shell32/CDefView.cpp

```diff
@@ -2311,46 +2342,74 @@ LRESULT CDefView::OnChangeNotify(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &
         return FALSE;
     }
 
-    BOOL bParent0 = ILIsParentOrSpecialParent(m_pidlParent, Pidls[0]);
-    BOOL bParent1 = ILIsParentOrSpecialParent(m_pidlParent, Pidls[1]);
-
     TRACE("(%p)(%p,%p,0x%08x)\n", this, Pidls[0], Pidls[1], lParam);
 
+    // Translate PIDLs.
+    // SHSimpleIDListFromPathW creates fake PIDLs (lacking some attributes)
```

Comment:
I think SHGetRealIDL does this for you?

Reply:
The current impl of `SHGetRealIDL` is rotten.

Reply:
Added.